advisor-ui/src/pages/VolunteerHours/VolunteerHours.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useContext, useState } from "react";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import axios from "axios";
+import { UserContext } from "../../UserContext.js";
+import { IconButton } from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { Link } from "react-router-dom";
+import "./VolunteerHours.css";
+
+export default function VolunteerHours() {
+  const [studentEvents, setStudentEvents] = useState([]);
+  const { user } = useContext(UserContext);
+
+  const fetchStudentEvents = async () => {
+    try {
+      const res = await axios.get(
+        `http://localhost:5000/api/student-event/${user.id}`,
+        { params: { studentId: user.id } }
+      );
+      const fetchedStudentEvents = res.data.studentEvents;
+      setStudentEvents(fetchedStudentEvents);
+    } catch (error) {
+      alert("Something went wrong!");
+    }
+  };
+
+  useEffect(() => {
+    fetchStudentEvents();
+  }, []);
+
+  const deleteStudentEventFromState = (id) => {
+    const filteredEvents = studentEvents.filter(
+      (studentEvent) => studentEvent.id !== id
+    );
+    setStudentEvents(filteredEvents);
+  };
+
+  const deleteStudentEventFromDB = async (id) => {
+    try {
+      const res = axios.delete(
+        `http://localhost:5000/api/student-event/${id}`,
+        { params: { studentEventId: id } }
+      );
+      return true;
+    } catch (error) {
+      alert("Unable to delete. Try again later.");
+      return false;
+    }
+  };
```

Comment:
I would define `deleteStudentEventFromDB` outside of the scope of the actual react component. I don't think it really matters in this case, but if this component were being rendered frequently, then the function would be redefined repeatedly as well, even though it's static.